internal/admin/handlers_api.go

```
@@ -2315,13 +2388,31 @@ func (s *Server) apiSettingsPatch(w http.ResponseWriter, r *http.Request) {
-			next.CustomEndpoints = *p.CustomEndpoints
-		} else if p.CustomEndpointsText != nil {
-			next.CustomEndpoints = splitCustomEndpointsText(*p.CustomEndpointsText)
+		// Both forms report under `customEndpoints`: the array form wins
+		// when both are sent, so reporting the textarea form under its own
+		// key would name a field that did not decide the outcome.
+		// Validate() below may still prune invalid entries, which is why
+		// the compare is against the pre-normalisation list — it answers
+		// "did this request ask for a different list", not "is the stored
+		// list byte-identical to what you sent".
+		if p.CustomEndpoints != nil || p.CustomEndpointsText != nil {
+			want := next.CustomEndpoints
```

golangci-lint (2.12.2)

[error] ineffectual assignment to want

(ineffassign)